drv/sidecar-front-io/src/leds.rs

```diff
@@ -0,0 +1,320 @@
+    },
+    // System
+    LedLocation {
+        controller: LedController::Left,
+        output: 23,
+    },
+];
+
+impl Leds {
+    pub fn new(
+        left_controller: &I2cDevice,
+        right_controller: &I2cDevice,
+    ) -> Self {
+        Self {
+            controllers: [
+                Pca9956B::new(left_controller),
+                Pca9956B::new(right_controller),
+            ],
+        }
+    }
+
+    /// Set the current to whatever DEFAULT_LED_CURRENT is
+    pub fn initialize_current(&self) -> Result<(), Error> {
+        self.set_current(DEFAULT_LED_CURRENT)?;
+
+        Ok(())
+    }
+
+    /// Set the current to `value`
+    pub fn set_current(&self, value: u8) -> Result<(), Error> {
+        for (_i, controller) in self.controllers.iter().enumerate() {
+            controller.set_iref_all(value)?;
+        }
+
+        Ok(())
+    }
+
+    /// Turns on the System LED to a PWM value of DEFAULT_LED_PWM
+    pub fn turn_on_system_led(&self) -> Result<(), Error> {
+        self.controllers[LED_MAP[SYSTEM_LED_IDX].controller as usize]
+            .set_a_led_pwm(LED_MAP[SYSTEM_LED_IDX].output, DEFAULT_LED_PWM)?;
+
+        Ok(())
+    }
+
+    /// Takes a `mask` of which ports need their LEDs turned on
+    pub fn update_led_state(&self, mask: u32) -> Result<(), Error> {
+        let mut data_l: [u8; 16] = [0; 16];
+        let mut data_r: [u8; 16] = [0; 16];
+
+        for i in 0..32 {
+            let bit_mask: u32 = 1 << i;
+            let pwm_value = if (mask & bit_mask) != 0 {
+                DEFAULT_LED_PWM
+            } else {
+                0
+            };
+
+            if LED_MAP[i].controller == LedController::Left {
+                data_l[LED_MAP[i].output as usize] = pwm_value;
+            } else {
+                data_r[LED_MAP[i].output as usize] = pwm_value;
+            }
+        }
+
+        self.controllers[LedController::Left as usize]
+            .set_all_led_pwm(&data_l)?;
+        self.controllers[LedController::Right as usize]
+            .set_all_led_pwm(&data_r)?;
+
+        Ok(())
+    }
+
+    /// Query device registers and return a summary of observed errors
+    pub fn error_summary(&self) -> Result<FullErrorSummary, Error> {
+        // Errors are being temporarily suppressed here due to a miswiring of
+        // the I2C bus at the LED controller parts. They will not be accessible
+        // without rework to older hardware, and newer (correct) hardware will
+        // be replacing the hold stuff very soon.
+        // TODO: remove error suppression here once Rev B hardware is in
+        let summary = FullErrorSummary {
+            left: match self.controllers[LedController::Left as usize]
+                .check_errors()
+            {
+                Ok(errs) => errs,
+                Err(_) => LedErrSummary {
+                    ..Default::default()
+                },
+            },
+            right: match self.controllers[LedController::Right as usize]
```

Comment:
After seeing this `as usize` cast a bunch, I think we should add a helper function

```rust
fn controller(&self, c: LedController) -> &Pca9956B {
    &self.controllers[c as usize]
}
```
and use it everywhere